src/atomic.rs

```diff
@@ -958,6 +958,38 @@ impl<'scope, T> Ptr<'scope, T> {
         self.as_raw().as_ref()
     }
 
+    /// Drops and deallocates the pointee.
+    ///
+    /// Calling this method on `p` is equivalent to:
+    ///
+    /// ```no_compile
+    /// drop(Box::from_raw(p.as_raw() as *mut T));
```

Comment:
Someone might be asking themselves:
*Why do we allocate objects using `Owned`, but then drop them using `Box`*?

I wonder if we should write this instead to avoid confusion:

```rust
/// drop(Owned::from_raw(p.as_raw() as *mut T));
```

Reply:
And that gets me thinking: We have `Owned::into_ptr` and `Ptr::destroy`, which are symmetrical functions going in opposite directions.

Should we perhaps name them `Owned::into_ptr` and `Ptr::into_owned`?

Reply:
I like this idea, because when you have a `Ptr` you can use `into_owned`, and when you have a pointer you can use `drop(Owned::from_ptr(..))`. Then we could indeed remove `defer_drop`. Moreover, it clearly states that we are retaking ownership of the pointer and the right to destroy it in the future.

Reply:
`Ptr::into_owned` sounds way better imho. This is probably enough though `scope.defer(move || p.into_owned());`, right? Looks fairly compact.

Reply:
@arthurprs 
```rust
scope.defer(move || drop(p.into_owned())); // OK
scope.defer(move || { p.into_owned(); }); // OK
scope.defer(|| drop(p.into_owned())); // Compiles, but I strongly encourage *always* using move closures.
scope.defer(move || p.into_owned()); // Error: expected (), found struct `atomic::Owned`
```